addons/binding/org.openhab.binding.smappee/src/main/java/org/openhab/binding/smappee/handler/SmappeeHandler.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.binding.smappee.handler;
+
+import static org.openhab.binding.smappee.SmappeeBindingConstants.*;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.smappee.service.InvalidConfigurationException;
+import org.openhab.binding.smappee.service.ReadingsUpdate;
+import org.openhab.binding.smappee.service.SmappeeDeviceReading;
+import org.openhab.binding.smappee.service.SmappeeService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SmappeeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Niko Tanghe - Initial contribution
+ */
+public class SmappeeHandler extends BaseThingHandler implements ReadingsUpdate {
+
+    private Logger logger = LoggerFactory.getLogger(SmappeeHandler.class);
+    private SmappeeService smappeeService;
+
+    public SmappeeHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (smappeeService == null || smappeeService.isInitialized() == false) {
+            return;
+        }
+
+        if (command instanceof RefreshType) {
+            try {
+                SmappeeDeviceReading readings = smappeeService.getDeviceReadings();
+                newState(readings);
+            } catch (InvalidConfigurationException invalidConfigurationException) {
+                updateStatus(ThingStatus.OFFLINE);
+            }
+        } else {
+            logger.debug("Command {} is not supported for channel: {}", command, channelUID.getId());
+        }
+    }
+
+    @Override
+    public void newState(SmappeeDeviceReading readings) {
+        updateState(CHANNEL_CONSUMPTION, new DecimalType(readings.getLatestConsumption()));
+        updateState(CHANNEL_SOLAR, new DecimalType(readings.getLatestSolar()));
+        updateState(CHANNEL_ALWAYSON, new DecimalType(readings.getLatestAlwaysOn()));
+    }
+
+    @Override
+    public void invalidConfig() {
+        updateStatus(ThingStatus.OFFLINE);

```

Comment:
It would be nice to set the reason. Use additional parameters about configuration error and possible error message hat will be presented to the user why the configuration is wrong, so the user has an idea what to fix.

Reply:
ok, but detailed config errors will be set in initialize()